packages/@glimmer/integration-tests/test/modifiers/on-test.ts

```diff
@@ -0,0 +1,458 @@
+import { castToBrowser, expect, HAS_NATIVE_PROXY } from '@glimmer/util';
+import { getInternalModifierManager } from '@glimmer/manager';
+import { on } from '@glimmer/runtime';
+
+import { jitSuite, RenderTest, test, click } from '../..';
+
+// check if window exists and actually is the global
+const hasDom =
+  typeof self === 'object' &&
+  self !== null &&
+  (self as Window['self']).Object === Object &&
+  typeof Window !== 'undefined' &&
+  self.constructor === Window &&
+  typeof document === 'object' &&
+  document !== null &&
+  self.document === document &&
+  typeof location === 'object' &&
+  location !== null &&
+  self.location === location &&
+  typeof history === 'object' &&
+  history !== null &&
+  self.history === history &&
+  typeof navigator === 'object' &&
+  navigator !== null &&
+  self.navigator === navigator &&
+  typeof navigator.userAgent === 'string';
+
+// do this to get around type issues for these values
+let global = window as any;
+const isChrome = hasDom
+  ? typeof global.chrome === 'object' && !(typeof global.opera === 'object')
+  : false;
+const isFirefox = hasDom ? typeof global.InstallTrigger !== 'undefined' : false;
+const isIE11 = !global.ActiveXObject && 'ActiveXObject' in window;
+
+interface Counters {
+  adds: number;
+  removes: number;
+}
+
+interface OnManager {
+  counters: Counters;
+  SUPPORTS_EVENT_OPTIONS: boolean;
+}
+
+function getOnManager() {
+  return (getInternalModifierManager(on) as unknown) as OnManager;
+}
+
+if (hasDom) {
+  class OnTest extends RenderTest {
+    static suiteName = '{{on}} Modifier';
+
+    startingCounters: Counters = { adds: 0, removes: 0 };
+
+    findElement(selector: string): Element {
+      return expect(
+        castToBrowser(this.element, 'div').querySelector(selector),
+        `BUG: expected to find ${selector}`
+      );
+    }
+
+    beforeEach() {
+      // might error if getOnManagerInstance fails
+      this.startingCounters = getOnManager().counters;
+    }
+
+    assertCounts(expected: Counters) {
+      this.assert.deepEqual(
+        getOnManager().counters,
+        {
+          adds: expected.adds + this.startingCounters.adds,
+          removes: expected.removes + this.startingCounters.removes,
+        },
+        `counters have incremented by ${JSON.stringify(expected)}`
+      );
+    }
+
+    @test
+    'SUPPORTS_EVENT_OPTIONS is correct (private API usage)'(assert: Assert) {
+      let { SUPPORTS_EVENT_OPTIONS } = getOnManager();
+
+      if (isChrome || isFirefox) {
+        assert.strictEqual(SUPPORTS_EVENT_OPTIONS, true, 'is true in chrome and firefox');
+      } else if (isIE11) {
+        assert.strictEqual(SUPPORTS_EVENT_OPTIONS, false, 'is false in IE11');
+      } else {
+        assert.expect(0);
```

Comment:
This assertion is failing CI because there is an `afterEach` assertion. I think this branch should just be removed.

```suggestion
```